tests/components/roku/conftest.py

```diff
@@ -87,8 +87,10 @@ async def init_integration(
     mock_roku: MagicMock,
 ) -> MockConfigEntry:
     """Set up the Roku integration for testing."""
-    mock_config_entry.unique_id = mock_device.info.serial_number
     mock_config_entry.add_to_hass(hass)
+    hass.config_entries.async_update_entry(
+        mock_config_entry, unique_id=mock_device.info.serial_number
+    )
```

Comment:
Shouldn't this be the other way around? First update and next add to hass?

Reply:
You can't update an entry that isn't added to hass as it will fail since its not registered yet

Reply:
It will blow up with 

```
> entry = self.data[entry_id]
KeyError: 'a9224633eefa6fea643a7d12fa2bedff'
```

We should probably have a guard to give a better exception in `async_update_entry`

```
diff --git a/homeassistant/config_entries.py b/homeassistant/config_entries.py
index e4cd2205671..421d424733a 100644
--- a/homeassistant/config_entries.py
+++ b/homeassistant/config_entries.py
@@ -1528,6 +1528,8 @@ class ConfigEntries:
         not fired and this function returns False
         """
         changed = False
+        if entry.entry_id not in self._entries:
+            raise UnknownEntry(entry.entry_id)
 
         if unique_id is not UNDEFINED and entry.unique_id != unique_id:
             # Reindex the entry if the unique_id has changed

```

Reply:
improved the error to match the other methods in https://github.com/home-assistant/core/pull/110023/commits/2d1a98c623c08efaa276550d10f8ad2b0d96c147